### PR TITLE
Fix some issues uncovered by 2.12.3

### PR DESCRIPTION
**Pull request by johnynek**
this addresses #640 cc @lrytz 

review @piyushnarang @ianoc ?

### CODE REVIEW

File: algebird-test/src/test/scala/com/twitter/algebird/CountMinSketchTest.scala [hunk 1]
```diff
@@ -930,7 +930,7 @@ class CMSFunctionsSpec extends PropSpec with PropertyChecks with Matchers {
   }
 
   property("throw IAE when deriving depth from invalid delta values") {
-    val invalidDeltas = Table("invalidDelta", 0.0, 1E-330, 1E-400)
```
Comment (johnynek) [inline algebird-test/src/test/scala/com/twitter/algebird/CountMinSketchTest.scala:933]: these numbers were invalid doubles and 2.12.3 won't parse them. scala 2.11 I think parses them to 0.0.

File: build.sbt [hunk 2]
```diff
@@ -79,7 +79,7 @@ val sharedSettings = scalariformSettings ++  Seq(
 
   scalacOptions ++= {
     if (scalaVersion.value startsWith "2.12")
-      Seq("-opt:l:classpath")
+      Seq("-opt-inline-from:com.twitter.algebird.**")
```
Comment (johnynek) [inline build.sbt:82]: make sure we are only inlining from the local project. I *think* this is the correct invocation.

**Comment by ianoc** [comment]
lg

File: build.sbt (hunk 2 above)
Comment (lrytz) [reply build.sbt:82]: you **also** need the `-opt:l:inline` flag, `-opt-inline-from` alone doesn't enable the optimizer, it just configures it.

**Pull request merged**